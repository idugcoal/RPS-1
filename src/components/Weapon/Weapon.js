import React from 'react';

const Weapon = ({ name, onClick }) => (
  <div onClick={onClick}>
    {name}
  </div>
);

export default Weapon;
