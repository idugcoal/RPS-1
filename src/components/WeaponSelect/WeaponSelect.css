.weapons {
  display: flex;
  justify-content: center;
  cursor: pointer;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}
