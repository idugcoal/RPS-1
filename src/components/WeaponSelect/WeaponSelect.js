import React from 'react';
import Weapon from '../Weapon/Weapon';
import { ROCK, PAPER, SCISSORS } from '../../constants';
import style from './WeaponSelect.css';

const WeaponSelect = ({ onClick }) => (
  <div className={style.weaponSelect}>
    <Weapon name='ROCK' onClick={() => onClick(ROCK)} />
    <Weapon name='PAPER' onClick={() => onClick(PAPER)} />
    <Weapon name='SCISSORS' onClick={() => onClick(SCISSORS)} />
  </div>

);

export default WeaponSelect;
