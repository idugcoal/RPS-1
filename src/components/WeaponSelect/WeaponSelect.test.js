import React from 'react';
import { shallow } from 'enzyme';
import WeaponSelect from './WeaponSelect';
import Weapon from '../Weapon/Weapon';

test('should render correctly', () => {
  const wrapper = shallow(<WeaponSelect />);
  expect(wrapper).toMatchSnapshot();
});

test('should call onClick', () => {
  const click = jest.fn();
  const wrapper = shallow(<WeaponSelect onClick={click} />);
  wrapper.find(Weapon).at(0).simulate('click');
  expect(click.mock.calls[0][0]).toBe(0);
});
