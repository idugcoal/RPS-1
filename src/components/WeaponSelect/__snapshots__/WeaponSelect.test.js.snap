// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    <Weapon
      name="ROCK"
      onClick={[Function]}
    />
    <Weapon
      name="PAPER"
      onClick={[Function]}
    />
    <Weapon
      name="SCISSORS"
      onClick={[Function]}
    />
  </div>
  <div>
     Choose Weapon 
  </div>
</div>
`;
